pyanaconda/modules/timezone/initialization.py

```diff
@@ -0,0 +1,118 @@
+#
+# Copyright (C) 2022  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+import time
+import requests
+
+from pyanaconda.timezone import get_preferred_timezone, is_valid_timezone
+from pyanaconda.core.configuration.anaconda import conf
+from pyanaconda.core.constants import NETWORK_CONNECTION_TIMEOUT
+from pyanaconda.modules.common.constants.services import NETWORK
+from pyanaconda.modules.common.structures.timezone import GeolocationData
+from pyanaconda.modules.common.task import Task
+from pyanaconda.modules.common.util import is_module_available
+
+from pyanaconda.anaconda_loggers import get_module_logger
+log = get_module_logger(__name__)
+
+
+class GeolocationTask(Task):
+    """Run geolocation"""
+    @property
+    def name(self):
+        return "Geolocate the system"
+
+    def run(self):
+        url = conf.timezone.geolocation_provider
+
+        log.info("Geoloc: starting lookup using provider: %s", url)
+        start_time = time.time()
+
+        if not self._wait_for_network():
+            log.error("Geoloc: no network connection")
+            return GeolocationData()
+
+        result = self._locate(url)
+        log.info(
+            "Geoloc: lookup finished in %1.1f seconds, got result: %s",
```

Comment:
In the original implementation we did not log the territory code and timezone to the normal logs that get stored on the installed system, but logged it into a "sensitive" log that was only stored to the installation environment for debugging purposes and not transferred to the target system.

This was done not to store on the machine the country the user was in when the machine was installed, which could in some circumstances be quite a sensitive information.

Given that we don't have such a sensitive log mechanism in place and given how DBus modules work it would not be very easy do I suggest not logging the result of the geolocation or at least not logging the territory code.

Reply:
Do you have a link to a discussion about that? I would like to point out that we dropped some really sensitive data like the public IP address and the city and we are keeping only the timezone and the territory code (which is actually the country code). The timezone is set as it is, so it is logged by the Timezone module and it was never protected anyway. The territory code is transformed to a locale and eventually logged by the Localization module. I am not sure if we really need to protect the `GB` territory code if we later happily log the `en_GB.UTF-8` locale and the `Europe/London` timezone. Anyway, we are not able to access these logs unless users attach them to a bug report or something and they can always set up the privacy. @jkonecny12, what do you think?

Reply:
I wonder what is the benefit of having this logged? Are we using it somewhere? If not I would avoid logging it because it's just safer and I don't think it's any use for us.

To answer the question. I see the difference between `en_GB.UTF-8` or `Europe/London` and data from geolocation service. The point is that the first two could be set by a user to anything. It doesn't necessarily means that you are in that country. However, data from geolocation are given from outside and it's strong point for location of the user.

Honestly, I don't think we should log any of those values if we don't have a strong reason to do that. We shouldn't do that just because we can. I don't think that `en_GB.UTF-8` is that much helpful in case of application crash. If it will crash because of the value you should see that in the traceback or you can ask user. Feel free to give here arguments for logging these :).

Reply:
I agree with not logging the result data, instead we can only log if the result was useful or not. The values were useful to debug this during development, but they are not really needed.

However - can we define the threat? If it is _another party can learn that when installing, the computer was geolocated in \<country-sized area>_, then even logging that any valid geolocation response happened is bad. Above, we agreed that we can tell from the other logs what values were set later. That means anyone else could, too. Is it enough to make it harder to get the data, or do we have to go all the way to make it impossible?